src/EditorFeatures/CSharp/SplitComment/SplitCommentCommandHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.Editor.Implementation.SplitComment;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.VisualStudio.Commanding;
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
+using Microsoft.VisualStudio.Text.Operations;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.SplitComment
+{
+    [Export(typeof(ICommandHandler))]
+    [ContentType(ContentTypeNames.CSharpContentType)]
+    [Name(nameof(SplitCommentCommandHandler))]
+    [Order(After = PredefinedCompletionNames.CompletionCommandHandler)]
+    internal partial class SplitCommentCommandHandler : AbstractSplitCommentCommandHandler
+    {
+        [ImportingConstructor]
+        public SplitCommentCommandHandler(
+            ITextUndoHistoryRegistry undoHistoryRegistry,
+            IEditorOperationsFactoryService editorOperationsFactoryService)
+        {
+            _undoHistoryRegistry = undoHistoryRegistry;
+            _editorOperationsFactoryService = editorOperationsFactoryService;
+        }
+
+        public override CommandState GetCommandState(ReturnKeyCommandArgs args)
+        {
+            return CommandState.Unspecified;
+        }
+
+        public override bool ExecuteCommand(ReturnKeyCommandArgs args, CommandExecutionContext context)
+        {
+            return ExecuteCommandWorker(args);
+        }
+
+        protected override bool LineContainsComment(ITextSnapshotLine line, int caretPosition)
+        {
+            var snapshot = line.Snapshot;
+            for (int i = line.Start; i < caretPosition; i++)
+            {
+                if (snapshot[i] == '/' && snapshot[i + 1] == '/')
```

Comment:
couldn't `i+1` index out of bounds when you're at the end of the file?

(this is worthwhile to test for as well).

Reply:
is this just asking "does this line contain `//`"?  If so, you could just ask for that.

Reply:
True. I’ll update